src/test/java/co/rsk/federate/bitcoin/BtcTransactionBuilder.java

```diff
@@ -0,0 +1,112 @@
+package co.rsk.federate.bitcoin;
+
+import co.rsk.bitcoinj.core.Address;
+import co.rsk.bitcoinj.core.BtcTransaction;
+import co.rsk.bitcoinj.core.Coin;
+import co.rsk.bitcoinj.core.NetworkParameters;
+import co.rsk.bitcoinj.core.TransactionInput;
+import co.rsk.bitcoinj.core.TransactionOutPoint;
+import co.rsk.bitcoinj.core.TransactionOutput;
+import co.rsk.bitcoinj.core.TransactionWitness;
+import co.rsk.bitcoinj.script.Script;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public final class BtcTransactionBuilder {
+
+    private NetworkParameters networkParameters = NetworkParameters.fromID(NetworkParameters.ID_MAINNET);
+    private final Map<Integer, TransactionWitness> transactionWitnesses = new HashMap<>();
+    private final List<TransactionInput> inputs = new ArrayList<>();
+    private final List<TransactionOutput> outputs = new ArrayList<>();
+
+    public InputBuilder createInputBuilder() {
+        return new InputBuilder();
+    }
+
+    public BtcTransactionBuilder withNetworkParameters(NetworkParameters networkParameters) {
+        this.networkParameters = networkParameters;
+        return this;
+    }
+
+    public BtcTransactionBuilder withInput(TransactionInput transactionInput) {
+        inputs.add(transactionInput);
+        return this;
+    }
+
+    public BtcTransactionBuilder withInputFromOutput(TransactionOutput transactionOutput) {
+        inputs.add(new InputBuilder().withAmount(transactionOutput.getValue()).withOutpointIndex(
+            transactionOutput.getIndex()).build());
+        return this;
+    }
+
+    public BtcTransactionBuilder withWitness(int inputIndex,
```

Comment:
check sonar suggestion here, is saying this method could return void since the return value is not being used